tests/GenericKMSWrapperTest.php

```diff
@@ -119,6 +119,41 @@ public function testRetryEncrypt()
         self::assertEquals($secret, $mockWrapper->decrypt($encrypted));
     }
 
+    public function testRetryEncryptFail()
+    {
+        $mockKmsClient = self::getMockBuilder(KmsClient::class)
+            // Need to pass service explicitly, because AwsClient fails to detect it from mock
+            ->setConstructorArgs([['region' => AWS_DEFAULT_REGION, 'version' => '2014-11-01', 'service' => 'kms']])
+            ->setMethods(['execute'])
+            ->getMock();
+        $callNo = 0;
```

Comment:
Nestačilo v obou tech nových testech prostě jenom pořád vyhazovat exception?
Vyzkošel jsem to a funguje to i tak ok.

```suggestion
        $mockKmsClient->method('execute')
            ->willReturnCallback(function (CommandInterface $command) use ($mockKmsClient) {
                throw new ConnectException('mock failed to connect', new Request('GET', 'some-uri'));
            });
```

Reply:
no jasne, stacilo :man_facepalming: